tests/compile-fail/arithmetic.rs

```diff
@@ -0,0 +1,31 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(integer_arithmetic, float_arithmetic)]
+#![allow(unused, shadow_reuse, shadow_unrelated, no_effect)]
+fn main() {
+    let i = 1i32;
+    1 + i; //~ERROR integer arithmetic detected
+    i * 2; //~ERROR integer arithmetic detected
+    1 % //~ERROR integer arithmetic detected
+    i / 2; 
+    i - 2 + 2 - i; //~ERROR integer arithmetic detected
+    -i; //~ERROR integer arithmetic detected
+    
+    i & 1; // no wrapping
+    i | 1; 
+    i ^ 1;
+    i % 7;
+    i >> 1;
+    i << 1;
```

Comment:
Ah, indeed, looks like it’s defined to be `255u8 << 1 == 254` and `255u8 * 2` ⇒ panic.